### PR TITLE
gh-85453: Fix 'time zone' spelling issues across datetime.rst

### PR DESCRIPTION
**Pull request by uatach**
Another pull request based on PR #21447, this one focused on fixing spelling issues on occurrences of 'time zone'. 

----
📚 Documentation preview 📚: https://cpython-previews--118449.org.readthedocs.build/

* Issue: gh-85453

### CODE REVIEW

File: Doc/library/datetime.rst [hunk 1]
```diff
@@ -1311,22 +1311,22 @@ Instance methods:
...
 
    If ``self.tzinfo`` is *tz*, ``self.astimezone(tz)`` is equal to *self*:  no
    adjustment of date or time data is performed. Else the result is local
-   time in the timezone *tz*, representing the same UTC time as *self*:  after
+   time in the time zone *tz*, representing the same UTC time as *self*:  after
    ``astz = dt.astimezone(tz)``, ``astz - astz.utcoffset()`` will have
    the same date and time data as ``dt - dt.utcoffset()``.
 
-   If you merely want to attach a time zone object *tz* to a datetime *dt* without
+   If you merely want to attach a :class:`timezone` object *tz* to a datetime *dt* without
```
Comment (uatach) [inline Doc/library/datetime.rst:1327]: @erlend-aasland there are similar issues for occurrences of `datetime` like this one, but I was not sure if I should include them in the same PR

**Comment by pganssle** [comment]
I think *timezone* is also an acceptable spelling, if that's what we've been consistently using, we should probably continue using that.

Alternatively, we could differentiate between `datetime.timezone` (the class) and the more abstract notion of a "time zone" by including the space for the abstract notion, but I'm more inclined to mark out references to the class with backticks / class references rather than a spelling difference.

**Comment by uatach** [comment]
> I think _timezone_ is also an acceptable spelling, if that's what we've been consistently using, we should probably continue using that.

I'm not sure... the IANA database uses *"time zone"* and both Wikipedia and The Free Dictionary redirect *"timezone"* to *"time zone"*... so maybe *"timezone"* isn't even valid spelling but people use it anyway?

> Alternatively, we could differentiate between `datetime.timezone` (the class) and the more abstract notion of a "time zone" by including the space for the abstract notion, but I'm more inclined to mark out references to the class with backticks / class references rather than a spelling difference.

Yes, I tried to differentiate all of them (the class and the instances and objects of the class) from the abstract concept.

**Comment by uatach** [comment]
Hello @erlend-aasland, any issues with this PR?

**Comment by erlend-aasland** [comment]
@uatach, I'll have a look in the next couple of days.

**Comment by erlend-aasland** [comment]
> Alternatively, we could differentiate between `datetime.timezone` (the class) and the more abstract notion of a "time zone" by including the space for the abstract notion, but I'm more inclined to mark out references to the class with backticks / class references rather than a spelling difference.

I think it would be an improvement to consistently use spell it "time zone" in the prose; it would help separate the class from the abstract concept even more.

Both spelling variants are used across the Python docs; I think we would benefit from a consistent spelling all across the docs. Moreover, most dictionaries seem to use "time zone", and as long as that is true, we're bound to get "spelling correction PRs" from time to time.

File: Doc/library/datetime.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/library/datetime.rst:1327]: If there are only a few, I think it is ok to include them here.

File: Doc/library/datetime.rst [hunk 2]
```diff
@@ -1311,22 +1311,22 @@ Instance methods:
...
    adjustment of date and time data, use ``dt.replace(tzinfo=tz)``. If you
-   merely want to remove the time zone object from an aware datetime *dt* without
+   merely want to remove the :class:`timezone` object from an aware datetime *dt* without
```
Comment (erlend-aasland) [inline Doc/library/datetime.rst]: We try to keep it at one unique link/reference pr. paragraph.

```suggestion
   merely want to remove the :class:`!timezone` object from an aware datetime *dt* without
```

File: Doc/library/datetime.rst [hunk 3]
```diff
@@ -2263,12 +2263,12 @@ only EST (fixed offset -5 hours), or only EDT (fixed offset -4 hours)).
     :mod:`zoneinfo`
       The :mod:`!datetime` module has a basic :class:`timezone` class (for
       handling arbitrary fixed offsets from UTC) and its :attr:`timezone.utc`
-      attribute (a UTC timezone instance).
+      attribute (a UTC :class:`timezone` instance).
```
Comment (erlend-aasland) [inline Doc/library/datetime.rst]:
```suggestion
      attribute (a UTC :class:`!timezone` instance).
```

File: Doc/library/datetime.rst (hunk 1 above)
Comment (uatach) [reply Doc/library/datetime.rst:1327]: I don't think there are only a few, I counted at least 15 and there are some "datetimes" that maybe should be replaced by "datetime objects" or "datetime instances"

**Review by erlend-aasland — APPROVED** [review]
Thanks; I believe this is an improvement. I'll await a few days before landing to give Paul time to chime in, in case he really don't want this to go in.

**Comment by nedbat** [comment]
I created a "time zone" entry in the style guide: https://github.com/python/devguide/pull/1352

**Comment by nedbat** [comment]
Should we also recommend "date and time", or is "datetime" better because it's compact and people will understand what we mean?

**Comment by erlend-aasland** [comment]
> Should we also recommend "date and time", or is "datetime" better because it's compact and people will understand what we mean?

With the "time zone" / "timezone" clarification in place, it would also make sense to clarify "datetime" / "date and time", IMO.

**Pull request merged**